src/main/java/com/cognifide/aemrules/htl/HtlSensor.java

```diff
@@ -0,0 +1,119 @@
+/*-
+ * #%L
+ * AEM Rules for SonarQube
+ * %%
+ * Copyright (C) 2015-2018 Cognifide Limited
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.cognifide.aemrules.htl;
+
+import com.cognifide.aemrules.htl.api.HtlCheck;
+import com.cognifide.aemrules.htl.lex.HtlLexer;
+import com.cognifide.aemrules.htl.visitors.DefaultHtlVisitor;
+import com.cognifide.aemrules.htl.visitors.HtlScanner;
+import java.io.File;
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import org.sonar.api.batch.Phase;
+import org.sonar.api.batch.fs.FilePredicate;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.InputFile.Type;
+import org.sonar.api.batch.rule.CheckFactory;
+import org.sonar.api.batch.sensor.Sensor;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.SensorDescriptor;
+import org.sonar.api.config.Configuration;
+import org.sonar.api.measures.FileLinesContextFactory;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.plugins.html.node.Node;
+import org.sonar.plugins.html.visitor.HtmlSourceCode;
+import org.sonar.squidbridge.ProgressReport;
+
+@Phase(name = Phase.Name.PRE)
+public class HtlSensor extends AbstractHtlFilesAnalyzer implements Sensor {
```

Comment:
I'm not a big fan of using abstraction and word "extends". It is much better to use Composition. 
Please treat this comment as my opinion not something that must be changed

Reply:
I second that, especially that we won't use this abstraction elsewhere, so this division is kind of artificial and arbitrary.

Reply:
renamed to HtlFilesAnalyzer